### ISSUE TITLE
Image_molecules not changing the trajectory

### ISSUE THREAD

**Issue by Alikusay** [mdtraj/mdtraj#1472]
I apologies in advance if I have missed something obvious, but I have failed to obtain a new trajectory with the protein centred after running Image_molecules(), please have a look at the code and let me know if I am missing something: 

```
import mdtraj as md
import os

top = md.load_pdb('Initial_charmm36_m.pdb')
traj = md.load_dcd('truncated.dcd',top=top)

#Ensuring that anchors are guessed to be protein atoms 
protein = traj.topology.guess_anchor_molecules()
newtraj = traj.image_molecules(anchor_molecules=protein)

newtraj.save_dcd('centered.dcd')
```

After inspecting the simulation, the issue persists and protein slowly diffuses from the centre. 

The image_molecules function returns the following errors multiple times, I am not sure if this is part of the issue: 

```
ValueError                                Traceback (most recent call last)
ValueError: Buffer dtype mismatch, expected 'int32_t' but got 'long'

Exception ignored in: 'mdtraj.geometry._geometry.image_frame'
ValueError: Buffer dtype mismatch, expected 'int32_t' but got 'long'
```

I have also tried passing the image_molecules function with "inplace=True" as well as running the function with a list of atoms from "chain 0" as discussed in issue: #1462 

Please also find attached file with truncated simulation and output files if needed as well images.
[Mdtraj_image_issue.tar.gz](https://github.com/mdtraj/mdtraj/files/3491202/Mdtraj_image_issue.tar.gz)

**Comment by fabian-paul** [comment]
I think this was fixed recently in #1466.

**Comment by marscher** [comment]
Thank you Fabian!
